### PR TITLE
Fixed Mobile Yoti Pathway + a number of tickets

### PR DESCRIPTION
**Pull request by gburnhill**

### CODE REVIEW

File: HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs [hunk 1]
```diff
@@ -26,17 +26,18 @@ public YotiController(IOptions<YotiOptions> options, IValidationService validati
...
         {
-            var viewModel = new AuthenticateViewModel { ClientSdkId = _options.ClientSdkId, DomId = _options.DomId, ScenarioId = _options.ScenarioId };
+            var viewModel = new AuthenticateViewModel { UserId = HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value, ClientSdkId = _options.ClientSdkId, DomId = _options.DomId, ScenarioId = _options.ScenarioId };
             return View(viewModel);
         }
 
-        [HttpPost]
-        public async Task<IActionResult> ValidateToken([FromBody] ValidateTokenRequest request, CancellationToken token = default)
+        [HttpGet]
+        public async Task<IActionResult> ValidateToken(string userId, string token, CancellationToken cancellationToken)
```
Comment (Goolyio) [inline HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs:34]: We shouldnt need to keep passing up the userid?

Comment (mike-f50) [reply HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs:34]: @gburnhill can clarify, but I think we have to pass it for the mobile journey.

**Comment by gburnhill** [comment]
Yes it needs to get the userid from the query as the HttpContext will not have it in the mobile journey (as opens a new tab).

File: HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs [hunk 2]
```diff
@@ -26,17 +26,18 @@ public YotiController(IOptions<YotiOptions> options, IValidationService validati
...
         {
-            var id = HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
-            if (id != null)
+            var id = userId ?? HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
+
+            if (id != null && token!= null)
```
Comment (Goolyio) [inline HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs:38]: If you look in the validation service it already does the token check and a few other checks

File: HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs (hunk 1 above)
Comment (gburnhill) [reply HelpMyStreetFE/HelpMyStreetFE/Controllers/YotiController.cs:34]: Unsurprisingly @Goolyio you were totally right about id

**Pull request merged**